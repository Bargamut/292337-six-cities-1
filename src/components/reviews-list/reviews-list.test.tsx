import * as React from 'react';
import * as renderer from 'react-test-renderer';

import ReviewsList from './reviews-list';

const mocks = {
  comments: [{
    id: 1,
    user: {
      id: 2,
      isPro: false,
      name: ``,
      avatarUrl: `4.jpg`
    },
    rating: 2,
    comment: `The room was spacious and clean.`,
    date: `2019-06-20T16:39:57.287Z`
  }, {
    id: 2,
    user: {
      id: 3,
      isPro: false,
      name: ``,
      avatarUrl: `5.jpg`
    },
    rating: 5,
    comment: `The pool looked nothing like the photos.`,
    date: `2019-06-21T17:39:57.287Z`
  }]
};

it(`Reviews List correctly renders after relaunch`, () => {
  const {comments} = mocks;

  const reviewsList = renderer.create(
    <ReviewsList
      id={0}
      isLoggedIn={false}
      reviews={comments}
    />
  ).toJSON();

  expect(reviewsList).toMatchSnapshot();
});
