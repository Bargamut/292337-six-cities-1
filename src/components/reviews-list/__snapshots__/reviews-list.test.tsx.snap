// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews List correctly renders after relaunch 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â·
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="4.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The room was spacious and clean.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-06-20T16:39:57.287Z"
        >
          June 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="5.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The pool looked nothing like the photos.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-06-21T17:39:57.287Z"
        >
          June 2019
        </time>
      </div>
    </li>
  </ul>
</section>
`;
